### ISSUE TITLE
S2589 & S2583 : False positive with string.IsNullOrEmpty()

### ISSUE THREAD

**Issue by DuAell** [SonarSource/sonar-dotnet#1432]
### Description

S2589 or S2583 are raised when using string.IsNullOrEmpty() after checking for null. But it's still possible for the value to be empty.

### Repro steps

```
class Program
    {
        static void Main(string[] args)
        {
            var data = Console.ReadLine();

            if (data == null) return;

            if (string.IsNullOrEmpty(data)) // False positive on "Change this condition so that it does not always evaluate to 'false'."
                DoSomething(); // Launched when data equals string.Empty. 
        }

        private static void DoSomething()
        {
            // My method
        }
    }

```
### Expected behavior

Those rules shouldn't be raised.

### Related information

* SonarQube Version 7.1 (build 11001)
* SonarC# Version 7.1 (build 5212)
* Visual Studio Version 4.6.01586
* If running through the Scanner for MSBuild, its version 3.1.0.1141
